api/modules/trips/views.py

```diff
@@ -205,3 +205,38 @@ def get_common_trips(request, user_id):
     serializer = TripSerializer(common_trips, many=True)
 
     return Response(serializer.data, status=status.HTTP_200_OK)
+
+
+@api_view(['GET'])
+def remove_user_from_trip(request, trip_id):
+    """
+    Dissociates current signed-in user from existing trip
+    :param request:
+    :param trip_id:
+    :return: 400 if user not present in the trip
+    :return: 404 if trip or user does not exist
+    :return: 200 successful
+    """
+    try:
+        trip = Trip.objects.get(pk=trip_id)
+
+        # if signed-in user not associated with requested trip
+        if request.user not in trip.users.all():
+            error_message = "User not a part of trip"
+            return Response(error_message, status=status.HTTP_400_BAD_REQUEST)
```

Comment:
Should be a `401`
Since the user is unauthorised to remove himself from the given trip